askama_parser/src/error.rs

```diff
@@ -0,0 +1,46 @@
+use nom::error::{Error, ErrorKind, FromExternalError, ParseError};
+use std::borrow::Cow;
+
+#[derive(Debug)]
+pub(crate) struct ParsingError<I> {
```

Comment:
We have `ParseError` already. How does this relate to that, why do we need to keep both?

Reply:
`ParserError` is the "general" error which always contains a message and doesn't contain any `nom` information. Well, I don't see any reason why not merging both so I'll modify this type and merge them.

Reply:
If we can merge them I think that would be good thing. I think maybe we could just name it `Error`. I think it should just live in `lib.rs` -- it's too small to get its own file for my taste.

Reply:
So merging both types would actually not be a great idea because of the generic argument, which have no use for users but that they would have to "live with". I can move the type to `lib.rs` though if you prefer.

Reply:
Hmm, okay, so what's the difference between both of these types?

Reply:
The public one only (and always) contains a string and is meant to be used for displaying an error whereas the private on contains `nom` information that we need at the end to generate `ParseError`'s message.

Reply:
I'll add a comment on the type to make it clear.